src/Compilers/CSharp/Portable/Binder/Semantics/Operators/OperatorKind.cs

```diff
@@ -657,5 +658,8 @@ internal enum BinaryOperatorKind
         LiftedBoolXor = Lifted | Bool | Xor,
         LiftedUserDefinedXor = Lifted | UserDefined | Xor,
         DynamicXor = Dynamic | Xor,
+
+        UserDefinedRange = UserDefined | Range,
+        LiftedUserDefinedRange = Lifted | UserDefined | Range,
```

Comment:
Huh, hadn't thought of that implication but yeah that's totally possible.

Reply:
And it was incredibly easy to implement! :blush: Just plumb `Lifted` in a handful of spots, and literally everything is already taken care of.

(See the tests that are already functioning)

Reply:
Note: will ikely need some IOperation testing here if you don't already have it.